cpu/cc2538/periph/gpio.c

```diff
@@ -213,10 +213,11 @@ static inline void handle_isr(uint8_t port_num)
     port->IC             = 0x000000ff;
     port->IRQ_DETECT_ACK = (0xff << (port_num * GPIO_BITS_PER_PORT));
 
-    for (int i = 0; i < GPIO_BITS_PER_PORT; i++) {
-        if (state & (1 << i)) {
-            isr_ctx[port_num][i].cb(isr_ctx[port_num][i].arg);
-        }
+    state &= (1 << GPIO_BITS_PER_PORT) - 1;
+    while (state) {
+        int pin = 8 * sizeof(state) - __builtin_clz(state) - 1;
```

Comment:
Is the 8 for 8 bit per `uint8_t`, is there a macro for this?

Reply:
Yea sorry that was silly.
spelling out the 32 bit from which the number of leading zeros is subtracted makes it much clearer that the result must be the position most significant set bit.